day1.scala

```diff
@@ -0,0 +1,26 @@
+// using scala 3.0.2
+
+package day1
+
+import scala.io.Source
+
+@main def part1(): Unit =
+  val input = Source.fromFile("input/day1").mkString
+  val answer = part1(input)
+  println(s"The solution is $answer")
+
+@main def part2(): Unit =
+  val input = Source.fromFile("input/day1").mkString
+  val answer = part2(input)
+  println(s"The solution is $answer")
+
+def part1(input: String): String = 
+  val depths = input.split('\n').iterator.map(_.toInt)
+  val pairs = depths.sliding(2).map(arr => (arr(0), arr(1)))
+  pairs.count((prev, next) => prev < next).toString
+
+def part2(input: String): String =
+  val depths = input.split('\n').iterator.map(_.toInt)
+  val sums = depths.sliding(3).map(arr => arr(0) + arr(1) + arr(2))
```

Comment:
Fun thing is that you don't actually need to sum everything, just compare element `i` with `i+3`

```suggestion
  val sums = depths.sliding(4).count(arr => arr(0) < arr(3))
```

Reply:
Oh that's crazy. It's a bit hard to understand but yes it is true.

I will explain that solution as a bonus in the website.

Reply:
I wrote it down altogether as:
```
val prev = arr(i) + arr(i+1) + arr(i+2)
val next = arr(i+1) + arr(i+2) + arr(i+3)
```
which made me realize that the two middle products are just not needed